executor/compiler.go

```diff
@@ -154,9 +157,48 @@ func (c *Compiler) Compile(ctx context.Context, stmtNode ast.StmtNode) (_ *ExecS
 			}
 		}
 	}
+	enabledPlanReplayerCapture := variable.EnablePlanReplayerCapture.Load()
+	failpoint.Inject("enablePlanReplayerCapture", func(val failpoint.Value) {
+		if val.(bool) {
+			enabledPlanReplayerCapture = true
+		}
+	})
+	if enabledPlanReplayerCapture {
+		checkPlanReplayerCaptureTask(c.Ctx, stmtNode)
+	}
+
```

Comment:
Can we move this operation into `planner/optimize.go:optimize()`? Putting it into the planner package seems more reasonable.

Reply:
To build a PlanReplayerDump task with less consuming in query process, we introducing the function in`exector`, thus we can't move it into planner which would cause cycle import.